api-testing/multicloud-api/src/main/java/com/opensds/utils/Utils.java

```diff
@@ -0,0 +1,116 @@
+package com.opensds.utils;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+/**
+ * This class used for common utils.
+ */
+public class Utils {
+
+    /**
+     * format strings for the date/time and date stamps required during signing
+     **/
+    public static final String ISO8601BasicFormat = "yyyyMMdd'T'HHmmss'Z'";
+
+    /**
+     * Get List of files
+     *
+     * @param beginPattern begin pattern (bucket) e.g: bucket_b237
+     * @param path folder path e.g: resource/inputs
+     * @return list of files
+     */
+    public static List<File> listFilesMatchingBeginsWithPatternInPath(final String beginPattern, String path) {
+        List<File> retFileList = new ArrayList<>();
+        try {
+            File dir = new File(path);
+            assert dir.isDirectory() : "Invalid directory path: "+path;
+            File[] files = dir.listFiles(new FilenameFilter() {
+                @Override
+                public boolean accept(File dir, String name) {
+                    return name.matches("^" + beginPattern + "+[a-z_1-9-]*.json");
+                }
+            });
+
+            for (File xmlfile : files) {
+                retFileList.add(xmlfile);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return retFileList;
+    }
+
+    /**
+     * Read file content.
+     *
+     * @param file file.
+     * @return file content.
+     */
+    public static String  readFileContentsAsString(File file) {
+        String content = null;
+        try {
+            content = new String(Files.readAllBytes(file.toPath()));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return content;
+    }
+
+    /**
+     * Get Bucket Name.
+     *
+     * @param bucketFile file path
+     * @return bucket name
+     */
+    public static String getBucketName(File bucketFile){
```

Comment:
Why this func name is getBucketName() ? Why cannot it be made generic..getFileNameFromDelim() or so.. You pass the file name and delimiter and get the result..Reusability

Reply:
I have renamed with getFileNameFromDelim

Reply:
Ok thanks